src/main/java/org/jabref/logic/ai/ingestion/IngestionService.java

```diff
@@ -28,7 +35,8 @@
  * Use this class in the logic and UI.
  */
 public class IngestionService {
-    private final Map<LinkedFile, ProcessingInfo<LinkedFile, Void>> ingestionStatusMap = new HashMap<>();
+    // We use a {@link TreeMap} here for the same reasons we use it in {@link ChatHistoryService}.
+    private final TreeMap<LinkedFile, ProcessingInfo<LinkedFile, Void>> ingestionStatusMap = new TreeMap<>((o1, o2) -> o1 == o2 ? 0 : o1.getLink().compareTo(o2.getLink()));
```

Comment:
You can use the more easier way with Compartor.comparing( ... ) 
https://www.baeldung.com/java-8-comparator-comparing